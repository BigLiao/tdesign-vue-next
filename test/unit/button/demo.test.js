import { mount } from '@vue/test-utils';

describe('Button', () => {
  it('base demo works fine', () => {
    const demo = require('@/examples/button/demos/base.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('block demo works fine', () => {
    const demo = require('@/examples/button/demos/block.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('icon demo works fine', () => {
    const demo = require('@/examples/button/demos/icon.vue').default;
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('loading demo works fine', () => {
    const demo = require('@/examples/button/demos/loading.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const demo = require('@/examples/button/demos/size.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('theme demo works fine', () => {
    const demo = require('@/examples/button/demos/theme.vue');
    const wrapper = mount(demo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
