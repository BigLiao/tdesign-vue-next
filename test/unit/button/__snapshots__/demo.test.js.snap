// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button base demo works fine 1`] = `
<div>
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-default"
    >
      主要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-dashed t-button--theme-default"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default"
    >
      文字按钮
    </button>
  </div>
</div>
`;

exports[`Button block demo works fine 1`] = `
<div
  class="demo-button-row"
>
  <div
    class="demo-button-col"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-default t-size-full-width"
    >
      主按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default t-size-full-width"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-dashed t-button--theme-default t-size-full-width"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default t-size-full-width"
    >
      文字按钮
    </button>
  </div>
</div>
`;

exports[`Button icon demo works fine 1`] = `
<div
  class="demo-button-icon"
>
  <div
    class="demo-button-row"
  >
    <div
      class="demo-button-col"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-default"
      >
        <svg
          class="t-icon t-icon-add"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M544 160L544 480 864 480 864 544 544 544 544 864 480 864 480 544 160 544 160 480 480 480 480 160 544 160Z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="t-button__text"
        >
          
        新建
      
        </span>
      </button>
       
      <button
        class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      >
        <svg
          class="t-icon t-icon-cloud-upload"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M468.50289536 179.12714432C608.79153792 154.39047104 745.2460352 247.01926976 794.2539584 391.6582944L794.2539584 391.6582944 807.2647168 395.574496C896.038208 425.3438656 960 509.20239488 960 608 960 731.7117824 859.7117824 832 736.128 831.9688384L736.128 831.9688384 288 832C164.28821632 832 64 731.7117824 64 608 64 497.83116352 143.53262016 406.23816576 248.32945216 387.5465056 279.1437664 281.07169152 360.97337664 198.08749952 468.50289536 179.12714432ZM729.1151104 399.78404736C686.1123968 289.70146816 582.09134912 224.08573824 479.61637888 242.15484032 399.60371328 256.263232 334.87417408 318.72093696 309.80669696 405.3383104L309.80669696 405.3383104 298.74383296 443.56458304 259.5672 450.55216384 248.90208576 452.83288128C178.74574144 470.36751488 128 533.90822464 128 608 128 696.3655616 199.63444032 768 288 768L288 768 736.123552 767.9688384 746.6312256 767.6310528C829.9972864 762.2332352 896 692.81808 896 608 896 535.53089472 847.3794624 472.63577984 778.5740992 453.709552L778.5740992 453.709552 744.6580736 444.71774528 733.6389056 412.19644352ZM544 383.68953792L679.7644992 519.46248384 634.50966784 564.71731776 575.97949824 506.24953792 576 704 512 704 511.97949824 506.12153792 453.49033216 564.70887424 408.23549824 519.45404032 544 383.68953792Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="t-button__text"
        >
          
        上传文件
      
        </span>
      </button>
    </div>
     
    <div
      class="demo-button-col"
    >
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-default t-button--icon-only t-button--shape-circle"
      >
        <svg
          class="t-icon t-icon-discount"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M493.04229568 116.1597152C508.86786752 116.30420928 524.0067104 121.97761344 535.39539904 133.36630208L942.6889024 540.659808C967.6824576 565.65336256 965.1497984 608.70858112 937.0320512 636.82633024L631.56192128 942.2964608C603.44417216 970.414208 560.3889536 972.9468672 535.39539904 947.953312L128.10189312 540.659808C116.179456 528.73737088 110.52056832 512.70502848 110.91453056 496.14468224L110.89530624 496.1568576 110.89530624 180.1597152C110.89530624 147.33822144 135.6018784 120.28725696 167.43155456 116.59028992L174.89530624 116.1597152 493.04229568 116.1597152ZM174.89530624 180.0957152L174.89530624 496.1568576 174.06330624 496.1277152 580.65023296 902.6984768 897.4340736 585.91464192 491.69530624 180.1597152 174.89530624 180.0957152ZM396.1453088 256.40588416C466.8377568 256.40588416 524.1453088 313.71343616 524.1453088 384.40588416 524.1453088 455.09833216 466.8377568 512.40588416 396.1453088 512.40588416 325.4528608 512.40588416 268.1453088 455.09833216 268.1453088 384.40588416 268.1453088 313.71343616 325.4528608 256.40588416 396.1453088 256.40588416ZM396.1453088 320.40588416C360.7990848 320.40588416 332.1453088 349.05966016 332.1453088 384.40588416 332.1453088 419.75210816 360.7990848 448.40588416 396.1453088 448.40588416 431.4915328 448.40588416 460.1453088 419.75210816 460.1453088 384.40588416 460.1453088 349.05966016 431.4915328 320.40588416 396.1453088 320.40588416Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </button>
       
      <button
        class="t-button t-size-m t-button--variant-base t-button--theme-default t-button--icon-only t-button--shape-circle"
      >
        <svg
          class="t-icon t-icon-cloud-download"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M468.50289536 179.12714432C608.79153792 154.39047104 745.2460352 247.01926976 794.2539584 391.6582944L794.2539584 391.6582944 807.2647168 395.574496C896.038208 425.3438656 960 509.20239488 960 608 960 731.7117824 859.7117824 832 736.128 831.9688384L736.128 831.9688384 288 832C164.28821632 832 64 731.7117824 64 608 64 497.83116352 143.53262016 406.23816576 248.32945216 387.5465056 279.1437664 281.07169152 360.97337664 198.08749952 468.50289536 179.12714432ZM729.1151104 399.78404736C686.1123968 289.70146816 582.09134912 224.08573824 479.61637888 242.15484032 399.60371328 256.263232 334.87417408 318.72093696 309.80669696 405.3383104L309.80669696 405.3383104 298.74383296 443.56458304 259.5672 450.55216384 248.90208576 452.83288128C178.74574144 470.36751488 128 533.90822464 128 608 128 696.3655616 199.63444032 768 288 768L288 768 736.123552 767.9688384 746.6312256 767.6310528C829.9972864 762.2332352 896 692.81808 896 608 896 535.53089472 847.3794624 472.63577984 778.5740992 453.709552L778.5740992 453.709552 744.6580736 444.71774528 733.6389056 412.19644352ZM576 384L575.97949824 581.696 634.50966784 523.28828096 679.7644992 568.54311488 544 704.2705088 408.23549824 568.50600576 453.49033216 523.25117184 511.97949824 581.76 512 384 576 384Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
     
    <div
      class="demo-button-col"
    >
      <button
        class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      >
        <svg
          class="t-icon t-icon-search"
          version="1.1"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 128C575.05800768 128 704 256.94199232 704 416 704 495.52900416 671.7644992 567.52900416 619.64675328 619.64675328L858.5096704 858.5096704 813.2548352 903.7644992 569.33373568 659.8328576C524.93400192 687.812288 472.35742272 704 416 704 256.94199232 704 128 575.05800768 128 416 128 256.94199232 256.94199232 128 416 128ZM416 192C292.28821632 192 192 292.28821632 192 416 192 539.71178368 292.28821632 640 416 640 539.71178368 640 640 539.71178368 640 416 640 292.28821632 539.71178368 192 416 192Z"
            fill-opacity=".9"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="t-button__text"
        >
          
        Function Icon
      
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Button loading demo works fine 1`] = `
<div
  class="button-base"
>
  <button
    class="t-button t-size-m t-button--variant-base t-button--theme-default t-is-loading"
  >
    <svg
      class="t-icon t-icon-loading"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 512C96 282.264 282.264 96 512 96 741.736 96 928 282.264 928 512L824 512 824 512C824 339.68715776 684.3128448 200 512 200 339.68715776 200 200 339.68715776 200 512 200 684.3128448 339.68715776 824 512 824L512 928C282.264 928 96 741.736 96 512Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="t-button__text"
    >
      
    主按钮
  
    </span>
  </button>
   
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default t-is-loading"
  >
    <svg
      class="t-icon t-icon-loading"
      version="1.1"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 512C96 282.264 282.264 96 512 96 741.736 96 928 282.264 928 512L824 512 824 512C824 339.68715776 684.3128448 200 512 200 339.68715776 200 200 339.68715776 200 512 200 684.3128448 339.68715776 824 512 824L512 928C282.264 928 96 741.736 96 512Z"
        fill-opacity=".9"
        fill-rule="evenodd"
      />
    </svg>
    <span
      class="t-button__text"
    >
      
    次按钮
  
    </span>
  </button>
</div>
`;

exports[`Button size demo works fine 1`] = `
<div>
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-s t-button--variant-base t-button--theme-default"
    >
      主按钮
    </button>
     
    <button
      class="t-button t-size-s t-button--variant-outline t-button--theme-default"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-s t-button--variant-dashed t-button--theme-default"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-s t-button--variant-text t-button--theme-default"
    >
      文字按钮
    </button>
  </div>
   
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-default"
    >
      主按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-dashed t-button--theme-default"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-default"
    >
      文字按钮
    </button>
  </div>
   
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-l t-button--variant-base t-button--theme-default"
    >
      主按钮
    </button>
     
    <button
      class="t-button t-size-l t-button--variant-outline t-button--theme-default"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-l t-button--variant-dashed t-button--theme-default"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-l t-button--variant-text t-button--theme-default"
    >
      文字按钮
    </button>
  </div>
</div>
`;

exports[`Button theme demo works fine 1`] = `
<div>
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-primary"
    >
      主要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-primary"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-dashed t-button--theme-primary"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-primary"
    >
      文字按钮
    </button>
  </div>
   
  <div
    class="tdesign-demo-block"
  >
    <button
      class="t-button t-size-m t-button--variant-base t-button--theme-danger"
    >
      主要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-danger"
    >
      次要按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-dashed t-button--theme-danger"
    >
      虚框按钮
    </button>
     
    <button
      class="t-button t-size-m t-button--variant-text t-button--theme-danger"
    >
      文字按钮
    </button>
  </div>
</div>
`;
